File: Exercism/c/nth-prime/src/nth_prime.c
#include <math.h>
#include <stdbool.h>
#include "nth_prime.h"

bool is_prime(int x) {
    if (x == 2 || x == 3)
        return true;

    if (x%2 == 0)
        return false;

    for (unsigned i = 3 ; i < (unsigned)sqrt(x)+1 ; i += 2) {
        if (x%i == 0)
            return false;
    }

    return true;
}

unsigned nth(unsigned num) {
    if (num < 1)
        return 0;

    unsigned nth = 0;

    for (int i = 2 ; ; ++i) {
        if (is_prime(i))
            nth++;

        if (nth == num)
            return i;
    }

    return num;
}

File: Exercism/dart/nth-prime/lib/nth_prime.dart
import 'dart:math';

class NthPrime {
  int prime(int num) {
    if (num < 1) {
      throw ArgumentError("There is no zeroth prime");
    }

    int counter = 0;
    int last_prime = 2;
    int idx = 2;

    while (counter != num) {
      if (isPrime(idx)) {
        last_prime = idx;
        counter += 1;
      }
      idx += 1;
    }

    return last_prime;
  }

  bool isPrime(int num) {
    if (num == 2 || num == 3)
      return true;

    if (num%2 == 0)
      return false;

    for (int i = 3 ; i <= sqrt(num).round() ; i += 2) {
      if (num%i == 0)
        return false;
    }

    return true;
  }
}
